simulation/halsim_ws_core/src/main/native/cpp/WSProvider_DriverStation.cpp

```diff
@@ -0,0 +1,164 @@
+    return {
+        {">autonomous", static_cast<bool>(HALSIM_GetDriverStationAutonomous())},
+    };
+  } else if (cbType == "Test") {
+    return {{">test", static_cast<bool>(HALSIM_GetDriverStationTest())}};
+  } else if (cbType == "EStop") {
+    return {{">estop", static_cast<bool>(HALSIM_GetDriverStationEStop())}};
+  } else if (cbType == "FmsAttached") {
+    return {{">fms", static_cast<bool>(HALSIM_GetDriverStationFmsAttached())}};
+  } else if (cbType == "DsAttached") {
+    return {{">ds", static_cast<bool>(HALSIM_GetDriverStationDsAttached())}};
+  } else if (cbType == "AllianceStationId") {
+    std::string station;
+    switch (HALSIM_GetDriverStationAllianceStationId()) {
+      case HAL_AllianceStationID_kRed1:
+        station = "red1";
+        break;
+      case HAL_AllianceStationID_kBlue1:
+        station = "blue1";
+        break;
+      case HAL_AllianceStationID_kRed2:
+        station = "red2";
+        break;
+      case HAL_AllianceStationID_kBlue2:
+        station = "blue2";
+        break;
+      case HAL_AllianceStationID_kRed3:
+        station = "red3";
+        break;
+      case HAL_AllianceStationID_kBlue3:
+        station = "blue3";
+        break;
+    }
+
+    return {{">station", station}};
+  } else if (cbType == "MatchTime") {
+    return {{"<match_time", HALSIM_GetDriverStationMatchTime()}};
+  }
+
+  return {};
+}
+
+void HALSimWSProviderDriverStation::OnNetValueChanged(const wpi::json& json) {
+  wpi::json::const_iterator it;
+  if ((it = json.find(">enabled")) != json.end()) {
+    HALSIM_SetDriverStationEnabled(it.value());
+  }
+  if ((it = json.find(">autonomous")) != json.end()) {
+    HALSIM_SetDriverStationAutonomous(it.value());
+  }
+  if ((it = json.find(">test")) != json.end()) {
+    HALSIM_SetDriverStationTest(it.value());
+  }
+  if ((it = json.find(">estop")) != json.end()) {
+    HALSIM_SetDriverStationEStop(it.value());
+  }
+  if ((it = json.find(">fms")) != json.end()) {
+    HALSIM_SetDriverStationFmsAttached(it.value());
+  }
+  if ((it = json.find(">ds")) != json.end()) {
+    HALSIM_SetDriverStationDsAttached(it.value());
+  }
+
+  if ((it = json.find(">station")) != json.end()) {
+    std::string station = it.value();
+    if (station == "red1") {
+      HALSIM_SetDriverStationAllianceStationId(HAL_AllianceStationID_kRed1);
+    } else if (station == "red2") {
+      HALSIM_SetDriverStationAllianceStationId(HAL_AllianceStationID_kRed2);
+    } else if (station == "red3") {
+      HALSIM_SetDriverStationAllianceStationId(HAL_AllianceStationID_kRed3);
+    } else if (station == "blue1") {
+      HALSIM_SetDriverStationAllianceStationId(HAL_AllianceStationID_kBlue1);
+    } else if (station == "blue2") {
+      HALSIM_SetDriverStationAllianceStationId(HAL_AllianceStationID_kBlue2);
+    } else if (station == "blue3") {
+      HALSIM_SetDriverStationAllianceStationId(HAL_AllianceStationID_kBlue3);
+    }
+  }
+
+  if ((it = json.find("joysticks")) != json.end()) {
```

Comment:
If you're moving to this new schema, it probably makes sense to have joysticks as separate providers? They are never called when the ds callback is called.